### ISSUE TITLE
Command-line error: language modes specified are incompatibleC/C++(1027)

### ISSUE THREAD

**Issue by ArturKrys** [microsoft/vscode-cpptools#10910]
### Environment

- OS and Version:  10.0.22621 N/A Build 22621
- WSL Version: Ubuntu 22.04.2 LTS
- VS Code Version: 1.77.3 (user setup)
- C/C++ Extension Version: v2.0.0

### Bug Summary and Steps to Reproduce

I am experiencing an issue with the C/C++ extension in VS Code where I receive the error "command-line error: language modes specified are incompatibleC/C++(1027)" for some C files. This error occurs even for files that have no code, but should compile without issue.

I am using VS Code on Windows and have the C/C++ extension installed. I write and save the C files in VS Code, but compile them on WSL Ubuntu. When compiling the same files on WSL Ubuntu, the files compile without errors.

I have tried uninstalling and reinstalling the C/C++ extension, as well as checking my workspace settings and file encoding. I have also disabled other extensions that may be causing conflicts.

Despite these efforts, the error persists. The issue occurs only on my Windows machine running VS Code and the C/C++ extension, and not on my Linux machine running the same version of VS Code with the same extensions.

Please let me know if there are any further steps I can take to resolve this issue, or if there is any additional information I can provide to assist in troubleshooting. Thank you for your help.

ps.: idk if this important but on the configurations c_cpp_properties.json it is giving this error:
Cannot find "/usr/bin/gcc".
Full path of the compiler being used, e.g. /usr/bin/gcc, to enable more accurate IntelliSense.

but i verified and it exists in the wsl

### Configuration and Logs

```shell
{
    "configurations": [
        {
            "name": "Linux",
            "includePath": [
                "${workspaceFolder}/**"
            ],
            "defines": [],
            "compilerPath": "/usr/bin/gcc",
            "cStandard": "c17",
            "cppStandard": "gnu++17",
            "intelliSenseMode": "linux-gcc-x64",
            "configurationProvider": "ms-vscode.makefile-tools"
        }
    ],
    "version": 4
}
```

### Other Extensions

i tried disabling everything expect wsl (v0.77.0) and C/C++extensions

**Comment by sean-mcmanus** [comment]
It sounds like you may be hitting https://github.com/microsoft/vscode-makefile-tools/issues/443 . Are you able to repro the issue after removing `"configurationProvider": "ms-vscode.makefile-tools"`?

**Comment by ArturKrys** [comment]
I apologize for my mistake. I have realized that I provided you with the wrong version of the C/C++ Extension Pack. The version I'm currently using is v1.3.0.

Regarding the issue, I appreciate your help, as it has partially resolved the problem. However, I am now encountering a peculiar error in my code. For instance, i have this line of code
printf("%d %.2f %.2f\n", 0, 0.0, 0.0);
The first argument, "0," triggers an error, indicating "too many arguments in function callC/C++(140)." I am confident that this issue is not related to the code itself since I am able to compile and run it without any problems.

(By the way, I would like to express my gratitude for your prompt response and how fast u were.)
